tedana/decomposition/ica.py

```diff
@@ -4,15 +4,18 @@
 import logging
 import warnings
 
+import sys
+
 import numpy as np
 from scipy import stats
 from sklearn.decomposition import FastICA
+from robustica import RobustICA ####BTBTBT
 
 LGR = logging.getLogger("GENERAL")
 RepLGR = logging.getLogger("REPORT")
 
 
-def tedica(data, n_components, fixed_seed, maxit=500, maxrestart=10):
+def tedica(data, n_components, fixed_seed, ica_method="robustica", n_robust_runs=30, maxit=500, maxrestart=10): ####BTBTBTB
```

Comment:
This is something I'd leave up to the TEDANA maintainers, but I'll lay the seed of the idea with you.
There's a large number of function keyword arguments here, and many look to be incompatible with one another. What might be a better long term solution would be to have a class that is responsible for storing an ICA configuration with respect to method, seeding, repeats, etc.. That class would have interface functions that would ensure that the configuration is always valid. An instance of that class could then be passed around the code.

Reply:
I agree with you and as you said we should leave it to the tedana team.